### PR TITLE
Add mexican regions

### PR DESCRIPTION
**Pull request by lexcast**
2.3 PR: https://github.com/magento/magento2/pull/21180

### Description (*)

Add mexican regions to magento
### Manual testing scenarios (*)

1. Go to checkout page
2. Select Mexico on address form
3. See mexican regions as options on region select

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18896) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @lexcast. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Directory/Setup/UpgradeData.php [hunk 1]
```diff
@@ -149,6 +152,49 @@ private function getDataForAustralia()
...
     }
 
+    /**
+     * Mexican states data.
+     *
+     * @return array
+     */
+    private function getDataForMexico()
+    {
+        return [
```
Comment (sivaschenko) [inline app/code/Magento/Directory/Setup/UpgradeData.php:162]: Hey @lexcast thanks for your contribution! I compared the region list with the wiki list and looks like "Mexico City" (CMX) region is missing https://en.wikipedia.org/wiki/Template:Mexico_State-Abbreviation_Codes However, I'm not sure if it should be included

File: app/code/Magento/Directory/Setup/UpgradeData.php [hunk 2]
```diff
@@ -149,6 +152,49 @@ private function getDataForAustralia()
...
+            'AGU' => 'Aguascalientes',
+            'BCN' => 'Baja California',
+            'BCS' => 'Baja California Sur',
+            'CAM' => 'Campeche',
+            'CHP' => 'Chiapas',
+            'CHH' => 'Chihuahua',
+            'CMX' => 'Ciudad de México',
```
Comment (lexcast) [inline app/code/Magento/Directory/Setup/UpgradeData.php:169]: Hi @sivaschenko, "Mexico City" is in this line, in spanish is "Ciudad de México"

**Comment by sivaschenko** [comment]
Hi @lexcast thank you for the update, we'd like to merge this contribution to 2.3-develop branch first. Could you please create a forwardport?

**Comment by osrecio** [comment]
Hi @sivaschenko I created a PR for 2.3 #21180 In order to merge this PR Soon for 2.2.

**Review by osrecio — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @osrecio, thank you for the review.
[ENGCOM-4240](https://jira.corp.magento.com/browse/ENGCOM-4240) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@lexcast thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by miguelbalparda** [comment]
This is quite interesting. For argentinian regions we have [a composer module](https://packagist.org/packages/mugar/module-argentina-regions), do you think is it worth including in the core also? Keep in mind any country might want to add regions too.

**Comment by osrecio** [comment]
Hi @miguelbalparda , IMHO all the countries should be welcome to the core. 

We can work to create a PR to add Regions of Argentina. What do you think? If it is declined we will have an aswer 😄

**Comment by miguelbalparda** [comment]
We are already working on [this](https://github.com/mug-ar/module-argentinaregions), will update you once all the work pending is completed.

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @lexcast can you please add missing commits including tests fix from the forward port

**Comment by sidolov** [comment]
@lexcast , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**